lib/improver/wind_direction.py

```diff
@@ -0,0 +1,404 @@
+        # containing ambigous data.
+        r_thresh = 0.01
+
+        if not isinstance(cube_ens_wdir, iris.cube.Cube):
+            msg = "Wind direction input is not a cube, but {}"
+            raise TypeError(msg.format(type(cube_ens_wdir)))
+
+        # Force input cube to float32.
+        enforce_float32_precision(cube_ens_wdir)
+
+        # Creates cubelists to hold data.
+        wdir_cube_list = iris.cube.CubeList()
+        r_vals_cube_list = iris.cube.CubeList()
+        std_dev_cube_list = iris.cube.CubeList()
+
+        y_coord_name = cube_ens_wdir.coord(axis="y").name()
+        x_coord_name = cube_ens_wdir.coord(axis="x").name()
+        for slice_ens_wdir in cube_ens_wdir.slices(["realization",
+                                                    y_coord_name,
+                                                    x_coord_name]):
+            # Extract wind direction data.
+            wind_dir_deg = slice_ens_wdir.data
+            realization_axis = slice_ens_wdir.coord_dims('realization')[0]
+
+            # Copies input cube and remove realization dimension to create
+            # cubes for storing results.
+            slice_mean_wdir = next(slice_ens_wdir.slices_over("realization"))
+            slice_mean_wdir.remove_coord('realization')
+            slice_r_vals = slice_mean_wdir.copy()
+            slice_std_dev = slice_mean_wdir.copy()
+
+            # Convert wind direction from degrees to complex numbers.
+            wind_dir_complex = WindDirection.deg_to_complex(wind_dir_deg)
+
+            # Find the complex average -  which actually signifies a point
+            # between all of the data points in POLAR co-cordinates.
+            # NOT the average DEGREE ANGLE.
+            wind_dir_complex_mean = np.mean(wind_dir_complex,
+                                            axis=realization_axis)
+
+            # Convert complex average values to degrees to produce average
+            # wind direction.
+            wind_dir_deg_mean = WindDirection.complex_to_deg(
+                wind_dir_complex_mean)
+
+            # Find radius values for wind direction average.
+            r_vals = WindDirection.find_r_values(wind_dir_complex_mean)
+
+            # Calculate standard deviation from polar mean.
+            # ToDo: This will still need some further investigation.
+            #        This is will be the subject of another ticket.
+            polar_mean_std_dev = WindDirection.calc_polar_mean_std_dev(
+                wind_dir_complex, wind_dir_deg_mean, r_vals, r_thresh,
+                realization_axis)
+
+            # Finds any meaningless averages and subistuite with
+            # the wind direction taken from the first ensemble member.
+            wind_dir_deg_mean = WindDirection.wind_dir_decider(
+                wind_dir_deg, wind_dir_deg_mean, r_vals, r_thresh)
+
+            # Save data into cubes.
+            slice_mean_wdir.data = wind_dir_deg_mean
+            slice_r_vals.data = r_vals
```

Comment:
Just so you know you could combine this with line 352 but you don't have to change it now.
e.g. `slice_r_vals = slice_mean_wdir.copy(data=r_vals)`
Same idea for the line below.

Reply:
changed